### PR TITLE
Fix Cross-compile Linking Args

### PR DESCRIPTION
**Pull request by lenary**
I was seeing some path resolution issues when using the clang cross file on a system that had a system install of the risc-v toolchain. Debugging this further, it seems like we need to pass these toolchain arguments down to `<lang>_link_args` as well as `<lang>_args` so that the clang driver picks up the right libraries to use.

Additionally, meson v0.53.1 introduced some per-language linker specification options, so I'm protecting us from getting bitten by that by explicitly setting those to the same as the `ld` option.

### CODE REVIEW

**Comment by lenary** [comment]
Verified on morty that these arguments now mean the `clang` driver finds the right libgcc to link to.

File: generate-clang-meson-cross-file.sh [hunk 1]
```diff
@@ -70,7 +72,9 @@ strip = '${toolchain_dest}/bin/${toolchain_target}-strip'
 needs_exe_wrapper = true
 has_function_printf = false
 c_args = [${meson_cflags}]
+c_link_args = [${meson_cflags}]
 cpp_args = [${meson_cflags}]
+cpp_link_args = [${meson_cflags}]
```
Comment (luismarques) [inline generate-clang-meson-cross-file.sh:77]: This concerns me a bit. It's true that some compilation options also need to be passed as link options, but others don't make sense. Is there a reason for why `meson_cflags` will always contain options that make sense for both compilation and linking? If not, can't we further separate this?
BTW, can you specify which options were missing? That might shed some light on the best way forward.

Comment (lenary) [reply generate-clang-meson-cross-file.sh:77]: I've been very careful to ensure only toolchain-related flags end up in `meson_cflags`.

What is a toolchain-related flag? One that is needed to ensure the compiler always compiles in a way that is compatible with the toolchain, or to ensure it uses the toolchain. This list currently is
- `-march`
- `-mabi`
- `-mcmodel`
- `--gcc-toolchain` (so clang does lookup correctly)

Comment (luismarques) [reply generate-clang-meson-cross-file.sh:77]: I was already assuming that you had done a good job with such separation :-) My point was that I was concerned that, given the name `meson_cflags`, it wouldn't be clear in the future. This was coming from the perspective of CFLAGS and LDFLAGS naming, where the CFLAGS content isn't necessarily applicable to LDFLAGS, so I was concerned that it wouldn't be clear that you couldn't add stuff to `meson_cflags` that would also apply to the linking stage. If you think this isn't going to be a problem or that there isn't an easy way to address this potential pitfall, then that's good enough :-)

Comment (lenary) [reply generate-clang-meson-cross-file.sh:77]: Ah, I see where you're going with this. Maybe we should call it `meson_driver_flags`?

Comment (luismarques) [reply generate-clang-meson-cross-file.sh:77]: Sounds good!

**Pull request merged**